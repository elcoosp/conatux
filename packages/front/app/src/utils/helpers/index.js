export const pipe = {
  execAsync: (...funcs) => context =>
    funcs.reduce(async (ctx, func) => {
      await func(ctx)
      return ctx
    }, context)
}
