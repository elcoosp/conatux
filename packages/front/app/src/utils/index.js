export { default as testing } from './testing'
export { default as linkify } from './linkify'
