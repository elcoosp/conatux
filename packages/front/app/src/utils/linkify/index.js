import React from 'react'
import { Link } from 'react-router-dom'

export const labelize = (path, label) => ({
  path,
  label
})

export const fromRoutes = (routesTree, componentName) => {
  if (['label', 'path'].every(prop => routesTree.hasOwnProperty(prop))) {
    const { path, label } = routesTree
    const link = props => (
      <Link to={path} {...props}>
        {label}
      </Link>
    )
    link.displayName = componentName
    return link
  }

  return Object.entries(routesTree).reduce(
    (links, [routeName, nestedRoute]) => {
      const componentName =
        routeName.slice(0, 1).toUpperCase() + routeName.slice(1)
      links[componentName] = fromRoutes(nestedRoute, componentName)
      return links
    },
    {}
  )
}
