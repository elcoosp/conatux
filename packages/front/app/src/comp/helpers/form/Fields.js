import React from 'react'
import { LabelledField } from '@:/comp/helpers/form'

export const Username = () => (
  <LabelledField label="Username" type="text" name="username" />
)

export const Email = () => (
  <LabelledField label="Email" type="email" name="email" />
)

export const Password = () => (
  <LabelledField label="Password" type="password" name="password" />
)

export const PasswordConfirmation = () => (
  <LabelledField
    label="Password confirmation"
    type="password"
    name="passwordConfirmation"
  />
)
export default {
  Username,
  Email,
  Password,
  PasswordConfirmation
}
