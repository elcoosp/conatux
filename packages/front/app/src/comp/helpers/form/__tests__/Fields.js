import { desc, getFakeProps } from '@:/utils/testing'
import Fields from '../Fields'

const fieldNames = ['Username', 'Email', 'Password', 'PasswordConfirmation']

const [Username, Email, Password, PasswordConfirmation] = getFakeProps(
  __dirname,
  '../Fields'
)(...fieldNames)

const props = {
  Username,
  Email,
  Password,
  PasswordConfirmation
}

fieldNames.forEach(fieldName =>
  desc.compWithSnap(
    { [`Fields.${fieldName}`]: Fields[fieldName] },
    props[fieldName],
    () => {}
  )
)
