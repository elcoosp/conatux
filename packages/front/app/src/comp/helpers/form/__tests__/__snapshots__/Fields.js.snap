// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fields.Email /> should match snapshot 1`] = `
<LabelledField
  label="Email"
  name="email"
  type="email"
/>
`;

exports[`<Fields.Password /> should match snapshot 1`] = `
<LabelledField
  label="Password"
  name="password"
  type="password"
/>
`;

exports[`<Fields.PasswordConfirmation /> should match snapshot 1`] = `
<LabelledField
  label="Password confirmation"
  name="passwordConfirmation"
  type="password"
/>
`;

exports[`<Fields.Username /> should match snapshot 1`] = `
<LabelledField
  label="Username"
  name="username"
  type="text"
/>
`;
